### ISSUE TITLE
Not Compiling with new 1.31.0 Nightly

### ISSUE THREAD

**Issue by Vidallia** [SergioBenitez/Rocket#802]
I'm trying to compile using the latest nightly 1.31.0 the example given on the Rocket Getting Started page from the website on Windows 10

My Cargo.toml dependencies:
```
[dependencies]
rocket = "0.3.17"
rocket_codegen = "0.3.17"
```

And the program I'm trying to compile:
```
#![feature(plugin)]
#![plugin(rocket_codegen)]

extern crate rocket;

#[get("/")]
fn index() -> &'static str {
    "Hello, Rocket"
}

fn main() {
    rocket::ignite().mount("/", routes![index]).launch();
}
```
The error I got seems to be with the _ring_ crate:

```
   Compiling ring v0.11.0
error: failed to run custom build command for `ring v0.11.0`
process didn't exit successfully: `C:\Users\haxrh\Documents\RustProjects\Rocket-Practice\hello-rocket\target\debug\build\ring-f402b078bd393b99\build-script-build` (exit code: 101)
--- stdout
=C:: C:\Users\haxrh\Documents\RustProjects\Rocket-Practice\hello-rocket
ALLUSERSPROFILE: C:\ProgramData
APPDATA: C:\Users\haxrh\AppData\Roaming
ASL.LOG: Destination=file
CARGO: \\?\C:\Users\haxrh\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin\cargo.exe
CARGO_CFG_DEBUG_ASSERTIONS:
CARGO_CFG_PROC_MACRO:
CARGO_CFG_TARGET_ARCH: x86_64
CARGO_CFG_TARGET_ENDIAN: little
CARGO_CFG_TARGET_ENV: msvc
CARGO_CFG_TARGET_FAMILY: windows
CARGO_CFG_TARGET_FEATURE: fxsr,mmx,sse,sse2
CARGO_CFG_TARGET_HAS_ATOMIC: 16,32,64,8,cas,ptr
CARGO_CFG_TARGET_OS: windows
CARGO_CFG_TARGET_POINTER_WIDTH: 64
CARGO_CFG_TARGET_THREAD_LOCAL:
CARGO_CFG_TARGET_VENDOR: pc
CARGO_CFG_WINDOWS:
CARGO_FEATURE_DEFAULT: 1
CARGO_FEATURE_DEV_URANDOM_FALLBACK: 1
CARGO_FEATURE_USE_HEAP: 1
CARGO_HOME: C:\Users\haxrh\.cargo
CARGO_MAKEFLAGS: --jobserver-fds=__rust_jobserver_semaphore_1020828091 --jobserver-auth=__rust_jobserver_semaphore_1020828091
CARGO_MANIFEST_DIR: C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0
CARGO_MANIFEST_LINKS: ring-asm
CARGO_PKG_AUTHORS: Brian Smith <brian@briansmith.org>
CARGO_PKG_DESCRIPTION: Safe, fast, small crypto using Rust.
CARGO_PKG_HOMEPAGE:
CARGO_PKG_NAME: ring
CARGO_PKG_REPOSITORY: https://github.com/briansmith/ring
CARGO_PKG_VERSION: 0.11.0
CARGO_PKG_VERSION_MAJOR: 0
CARGO_PKG_VERSION_MINOR: 11
CARGO_PKG_VERSION_PATCH: 0
CARGO_PKG_VERSION_PRE:
COMMONPROGRAMFILES: C:\Program Files\Common Files
COMMONPROGRAMFILES(X86): C:\Program Files (x86)\Common Files
COMMONPROGRAMW6432: C:\Program Files\Common Files
COMPUTERNAME: DESKTOP-3KMOR7D
COMSPEC: C:\WINDOWS\system32\cmd.exe
DEBUG: true
DRIVERDATA: C:\Windows\System32\Drivers\DriverData
GOOGLE_DEFAULT_CLIENT_ID: no
GOOGLE_DEFAULT_CLIENT_SECRET: no
HOMEDRIVE: C:
HOMEPATH: \Users\haxrh
HOST: x86_64-pc-windows-msvc
LD_LIBRARY_PATH: C:\Users\haxrh\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib
LOCALAPPDATA: C:\Users\haxrh\AppData\Local
LOGONSERVER: \\DESKTOP-3KMOR7D
NUMBER_OF_PROCESSORS: 8
NUM_JOBS: 8
ONEDRIVE: C:\Users\haxrh\OneDrive
OPT_LEVEL: 0
OS: Windows_NT
OUT_DIR: C:\Users\haxrh\Documents\RustProjects\Rocket-Practice\hello-rocket\target\debug\build\ring-a4a729f4e99bcc95\out
PATH: C:\Users\haxrh\Documents\RustProjects\Rocket-Practice\hello-rocket\target\debug\deps;C:\Users\haxrh\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Users\haxrh\.cargo\bin;C:\Users\haxrh\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Python36-32\Scripts\;C:\Program Files (x86)\Python36-32\;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\PuTTY\;C:\Program Files\Git\cmd;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\nodejs\;C:\Program Files\Java\jdk-9.0.4\bin;C:\Program Files (x86)\OpenVPN\bin;C:\cygwin64\bin;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\CMake\bin;C:\Program Files (x86)\IncrediBuild;C:\Users\haxrh\.cargo\bin;C:\Users\haxrh\AppData\Local\Microsoft\WindowsApps;C:\Users\haxrh\AppData\Local\atom\bin;C:\Users\haxrh\AppData\Local\Microsoft\WindowsApps;C:\Users\haxrh\AppData\Roaming\npm;C:\Users\haxrh\AppData\Local\hyper\app-2.0.0\resources\bin;;C:\Users\haxrh\AppData\Local\Programs\Microsoft VS Code\bin;C:\Exercism
PATHEXT: .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC,.py;.PY;.PYW;.MSC
PROCESSOR_ARCHITECTURE: AMD64
PROCESSOR_IDENTIFIER: Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
PROCESSOR_LEVEL: 6
PROCESSOR_REVISION: 8e0a
PROFILE: debug
PROGRAMDATA: C:\ProgramData
PROGRAMFILES: C:\Program Files
PROGRAMFILES(X86): C:\Program Files (x86)
PROGRAMW6432: C:\Program Files
PROMPT: $P$G
PSMODULEPATH: C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
PUBLIC: C:\Users\Public
RUSTC: rustc
RUSTDOC: rustdoc
RUSTUP_HOME: C:\Users\haxrh\.rustup
RUSTUP_TOOLCHAIN: nightly-x86_64-pc-windows-msvc
RUST_RECURSION_COUNT: 1
SESSIONNAME: Console
SYSTEMDRIVE: C:
SYSTEMROOT: C:\WINDOWS
TARGET: x86_64-pc-windows-msvc
TEMP: C:\Users\haxrh\AppData\Local\Temp
TERM_PROGRAM: vscode
TERM_PROGRAM_VERSION: 1.28.2
TMP: C:\Users\haxrh\AppData\Local\Temp
USERDOMAIN: DESKTOP-3KMOR7D
USERDOMAIN_ROAMINGPROFILE: DESKTOP-3KMOR7D
USERNAME: haxrh
USERPROFILE: C:\Users\haxrh
VBOX_MSI_INSTALL_PATH: C:\Program Files\Oracle\VirtualBox\
VSCODE_CWD: C:\Users\haxrh\AppData\Local\Programs\Microsoft VS Code
WINDIR: C:\WINDOWS
OPT_LEVEL = Some("0")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
TARGET = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
CC_x86_64-pc-windows-msvc = None
HOST_CC = None
OPT_LEVEL = Some("0")
CC_x86_64_pc_windows_msvc = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST_CC = None
HOST = Some("x86_64-pc-windows-msvc")
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
TARGET = Some("x86_64-pc-windows-msvc")
CFLAGS = None
HOST = Some("x86_64-pc-windows-msvc")
DEBUG = Some("true")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\limbs.obj" "crypto/limbs/limbs.c"
DEBUG = Some("true")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\ecp_nistz256.obj" "crypto/ec/ecp_nistz256.c"
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\constant_time_test.obj" "crypto/constant_time_test.c"
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\generic.obj" "crypto/bn/generic.c"
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\gfp_p384.obj" "crypto/ec/gfp_p384.c"
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "include" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/WX" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/Od" "/RTCsu" "/W4" "-c" "/FoC:\\Users\\haxrh\\Documents\\RustProjects\\Rocket-Practice\\hello-rocket\\target\\debug\\build\\ring-a4a729f4e99bcc95\\out\\curve25519.obj" "crypto/curve25519/curve25519.c"
limbs.c
ecp_nistz256.c
constant_time_test.c
generic.c
gfp_p384.c
curve25519.c
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch
specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(127) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(47) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(93) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(47) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(46) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(47) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(73) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(58) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(93) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(89) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(93) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(73) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(72) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(75) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(58) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(57) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(58) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(111) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(110) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(111) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(126) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(125) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(126) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(143) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(142) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(143) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(138) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(137) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(143) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(161) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(160) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.c(161) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(117) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch
specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(116) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(117) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch
specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(127) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(224) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(224) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(223) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(224) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch
specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(127) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(128) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(117) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch
specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(116) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\limbs\limbs.inl(117) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(93) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(92) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\gfp_p384.c(93) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz384.inl(35) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz384.inl(34) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz384.inl(37) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(208) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(208) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(207) : note: index 'j' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(217) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(205) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(204) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(217) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(194) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(193) : note: index 'j' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(217) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(191) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(190) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(217) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(185) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(184) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\constant_time_test.c(217) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3510) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3510) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3509) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3510) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3495) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3494) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3495) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3685) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3669) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3685) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3682) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3669) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3682) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3677) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3669) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3677) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3674) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3669) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\curve25519\curve25519.c(3674) : note: feeds call on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz256.c(76) : error C2220: warning treated as error - no 'object' file generated
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz256.c(76) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz256.c(75) : note: index 'i' range checked by comparison on this line
c:\users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\crypto\ec\ecp_nistz256.c(78) : note: feeds call on this line

--- stderr
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
thread '<unnamed>' panicked at 'execution failed', C:\Users\haxrh\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.11.0\build.rs:658:9
```

**Comment by jebrosen** [comment]
Duplicate of several issues including #642. On Windows, you are (unfortunately) pretty much stuck with using `master` or waiting for the `0.4` release.

**Comment by Trolldemorted** [comment]
Can you estimate when 0.4 (or any other commit that does not have this problem) will be released?

**Comment by jebrosen** [comment]
> Can you estimate when 0.4 (or any other commit that does not have this problem) will be released?

It's been fixed in `master` for a while. A 0.4 release candidate (`0.4.0.rc-1`) was published to crates.io on 2018-10-31.

**Comment by Trolldemorted** [comment]
0.4.0-rc.1 doesn't work for me:

```
PS D:\repositories\foo> cargo build
    Updating crates.io index
  Downloaded rocket v0.4.0-rc.1
  Downloaded pear v0.1.2
  Downloaded rocket_codegen v0.4.0-rc.1
  Downloaded rocket_http v0.4.0-rc.1
  Downloaded devise v0.1.0
  Downloaded cookie v0.11.0
  Downloaded quote v0.6.10
  Downloaded devise_codegen v0.1.0
  Downloaded pear_codegen v0.1.2
  Downloaded proc-macro2 v0.4.24
  Downloaded devise_core v0.1.0
  Downloaded syn v0.15.21
  Downloaded ring v0.13.5
  Downloaded untrusted v0.6.2
error: multiple packages link to native library `ring-asm`, but a native library can be linked
only once

package `ring v0.13.5`
    ... which is depended on by `cookie v0.11.0`
    ... which is depended on by `rocket_http v0.4.0-rc.1`
    ... which is depended on by `rocket v0.4.0-rc.1`
    ... which is depended on by `enokey v0.1.1 (D:\repositories\ENOKEY)`
links to native library `ring-asm`

package `ring v0.11.0`
    ... which is depended on by `cookie v0.9.2`
    ... which is depended on by `rocket v0.3.17`
    ... which is depended on by `rocket_codegen v0.3.17`
    ... which is depended on by `enokey v0.1.1 (D:\repositories\ENOKEY)`
also links to native library `ring-asm`
```

**Comment by jebrosen** [comment]
See the CHANGELOG, especially this section: https://github.com/SergioBenitez/Rocket/blob/v0.4/CHANGELOG.md#codegen-rewrite
